src/components/link/Link.tsx

```diff
@@ -0,0 +1,41 @@
+import Image from 'next/image'
+
+type Icon = {
+  leftIcon?: string
+  rightIcon?: JSX.Element
```

Comment:
I would have expected the two `icons` to be of the same type and to be handled the same way in the `jsx` rendering.

This makes `props` contract a bit confusing.